Automattic/wp-calypso
# Deprecate `useIsCustomDomainEnabledForFreePlan` and `useIsPlanUpsellEnabledOnFreeDomain` hooks

## Proposed Changes

This PR removes `useIsCustomDomainEnabledForFreePlan` and `useIsPlanUpsellEnabledOnFreeDomain` hooks. These two hooks were first introduced for fetching the corresponding ExPlat experiments data, and have been kept in case we want to introduce similar tests. That's why they returns hard-coded `DataResponse<>` values while there is no real request made underneath. However, now that there is no such an experiment planned, they should be replaced by simple values to reduce code complexity.

## Testing Instructions



In `/start/`,

* Choose a paid domain and then choose the Free plan -> the "a paid plan is required" modal should appear:

<img width="689" alt="CleanShot 2023-12-18 at 18 03 51@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/efc6c9fa-8385-4852-80bc-858c497a6510">

* Choose a free domain and then choose the Free plan -> there shouldn't be any modal.

In `/start/onboarding-pm`:

* Choose a paid domain and then choose the Free plan -> the "your custome domain will be a redirect" modal should appear:

<img width="736" alt="CleanShot 2023-12-18 at 18 04 55@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/845b378d-2f9d-4271-a73f-84199fcf3ab1">

* Choose a free domain and then choose the Free plan -> the "don't miss out" modal should appear.

<img width="669" alt="CleanShot 2023-12-18 at 18 04 34@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/7de60053-8bf4-4b94-b11a-e788dbd5d095">

## Pre-merge Checklist
